breadcord/bot.py

```diff
@@ -153,11 +153,20 @@ async def setup_hook(self) -> None:
         _logger.debug(f'Module search paths: {search_paths}')
         self.modules.discover(self, search_paths=search_paths)
 
+        failed = []
         for module in self.settings.modules.value:
             if module not in self.modules:
                 _logger.warning(f"Module '{module}' enabled but not found")
                 continue
-            await self.modules.get(module).load()
+            try:
+                await self.modules.get(module).load()
+            except Exception as error:
+                _logger.exception(f'Failed to load module {module!r}: {error}')
+                # Not needed as of writing (2024/03/29), but it means we won't ever have a "ghost loaded" module
+                self.modules.get(module).loaded = False
+                failed.append(module)
+        if failed:
+            _logger.warning('Failed to load modules: ' + ', '.join(f'{module}' for module in failed))
```

Comment:
Why `f'{module}'`, isn't `module` already a string?

Reply:
It *should* be, but I have a habit of being paranoid when it comes to handling error cases

Reply:
Given it's already being used a handful of times above under the assumption that it's a string, I think it's safe to just use `', '.join(failed)` - alongside a quick test in a debugger if you're paranoid

Reply:
The setting value is not guaranteed to be a list of strings, right?
Line 166 does assume it's a string though, and will error if it's not.

Reply:
I believe there is very primitive type validation handled by the setter of the `config.Setting.value` property, which ensures that the "outermost type" matches - i.e. `self.settings.modules` is guaranteed to be a list otherwise [a ValueError is thrown](https://github.com/Breadcord/Breadcord/blob/7a0f66fce2b52ad74da79ed80f488ea319667447/breadcord/config.py#L116-L118). However, if the end user decides to put a non-string type into the modules list, where the default value is already a list of strings, then honestly that's their fault. While validation for this would be nice, this is definitely not the place for it.